src/web/components/SubNav/SubNav.tsx

```diff
@@ -1,87 +1,207 @@
-import React, { Component, createRef } from 'react';
+import React, { useState, useRef, useEffect } from 'react';
+import { css, cx } from 'emotion';
 
 import { GuardianLines } from '@frontend/web/components/GuardianLines';
+import { text } from '@guardian/src-foundations/palette';
+import { palette } from '@guardian/src-foundations';
+import { textSans } from '@guardian/src-foundations/typography';
+import { from } from '@guardian/src-foundations/mq';
+import { pillarPalette, pillarMap } from '@root/src/lib/pillars';
 
-import { Inner } from './Inner';
-
-interface Props {
-    subnav?: SubNavType;
+type Props = {
+    subNavSections?: SubNavType;
     pillar: Pillar;
     currentNavLink: string;
-}
+};
+
+const wrapperCollapsed = css`
+    height: 36px;
+    overflow: hidden;
 
-export class SubNav extends Component<
-    Props,
-    {
-        showMore: boolean;
-        isExpanded: boolean;
+    ${from.tablet} {
+        height: 42px;
     }
-> {
-    private ulRef: React.RefObject<HTMLUListElement>;
-
-    constructor(props: Props) {
-        super(props);
-        this.state = { showMore: false, isExpanded: false };
-        // More should not be shown without JS
-        this.ulRef = createRef();
+`;
+
+const subnav = css`
+    list-style: none;
+    padding: 0 5px;
+
+    ${from.mobileLandscape} {
+        padding: 0 15px;
     }
 
-    public componentDidMount() {
-        // If componentDidMount runs we know client-side JS is enabled
-        this.setState({
-            showMore: this.shouldShowMore(),
-        });
+    li {
+        float: left;
+        display: block;
     }
+`;
 
-    public toggle = () => {
-        this.setState(prevState => ({
-            isExpanded: !prevState.isExpanded,
-        }));
-    };
-
-    public shouldShowMore() {
-        // get ul ref
-        const ul = this.ulRef.current;
-        if (!ul) {
-            return false;
-        }
+const subnavExpanded = css`
```

Comment:
Given we're in the SubNav component the subnav... prefix is redundant. Maybe `expandedStyles`?

Reply:
Also, we rarely use this composition method (this is the first time I've seen it in DCR) - it'd be great to refactor this to use `cx` instead for consistency